src/hotspot/share/jfr/periodic/sampling/jfrThreadSampler.cpp

```diff
@@ -386,6 +387,11 @@ bool JfrThreadSampleClosure::do_sample_thread(JavaThread* thread, JfrStackFrame*
     return false;
   }
 
+  // Process the oops in the thread head before calling into code that wants to
+  // stack walk over Loom continuations. The stack walking code will otherwise
+  // skip frames in stack chunks on the Java heap.
+  StackWatermarkSet::start_processing(thread, StackWatermarkKind::gc);
```

Comment:
does the "start_processing()" call ever reach a malloc, directly or indirectly? If so, it is problematic to call from the sampler thread.